[jit/docs] move migration guide to appendix

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31069 [jit/docs] add a warning for script classes
* **#31068 [jit/docs] move migration guide to appendix**

Let's get it out of the early parts now that the recursive API has been
around for a while

Differential Revision: [D18920498](https://our.internmc.facebook.com/intern/diff/D18920498)

suo committed:
[jit/docs] move migration guide to appendix

Let's get it out of the early parts now that the recursive API has been
around for a while

kostmo commented:
## CircleCI build failures summary
As of commit 2af0c9a:
* **4/4** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=2af0c9a5878717fec4a6e434b2933fe3a797af8a):

### 3 failures recognized by patterns:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3885640) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_build (1/3)
**Step:** Build ([details](https://dr.pytorch.org/build-details.html?build_id=71802179))
```
Dec 10 20:07:09 FAILED: caffe2/CMakeFiles/torch_cpu.dir/operators/lengths_reducer_fused_8bit_rowwise_ops.cc.o
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3885635) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build (2/3)
**Step:** Build ([details](https://dr.pytorch.org/build-details.html?build_id=71802931))
```
Dec 10 20:07:31 /pytorch/caffe2/operators/lengths_reducer_fused_8bit_rowwise_ops.h:61:55: error: expected primary-expression before ‘,’ token
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3885683) binary_linux_manywheel_2_7mu_cpu_devtoolset7_build (3/3)
**Step:** Build ([details](https://dr.pytorch.org/build-details.html?build_id=71803333))
```
Dec 10 20:08:03 /pytorch/caffe2/operators/lengths_reducer_fused_8bit_rowwise_ops.h:61:55: error: expected primary-expression before ‘,’ token
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

driazati reviewed (approved):
Looks fine but there’s some sub module updates

suo committed:
Update on "[jit/docs] move migration guide to appendix"


Let's get it out of the early parts now that the recursive API has been
around for a while
